# Tripal Crop Genetics

**Developed by the University of Saskatchewan, Pulse Crop Bioinformatics team.**

**NOTE: This package will replace the following Tripal v3 modules: [nd_genotypes](https://github.com/UofS-Pulse-Binfo/nd_genotypes), [genotypes_loader](https://github.com/UofS-Pulse-Binfo/genotypes_loader), [tripal_qtl](https://github.com/UofS-Pulse-Binfo/tripal_qtl), [vcf_filter](https://github.com/UofS-Pulse-Binfo/vcf_filter).**

- Genetic maps, markers, sequence variants and QTL
- Large-scale genotypic datasets with both 

    - the power of a relational database for tight integration with germplasm, phenotypic data and cross data type tools
    - the speed/ease of flat file storage and querying via the Variant Call Format (VCF)
 
- Genotype Matrix tool for quick visual querying of genotypic differences between germplasm in smaller regions (e.g. QTL or GWAS peak)
- Management of metadata for VCF files including a form for researchers to filter and download the results in multiple formats.
